src/main/java/frc/robot/commands/IntakeSequence.java

```diff
@@ -1,46 +1,44 @@
 package frc.robot.commands;
 
-import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+import edu.wpi.first.wpilibj2.command.ParallelRaceGroup;
+import edu.wpi.first.wpilibj2.command.RunCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.Cal;
 import frc.robot.subsystems.Intake;
 import frc.robot.subsystems.Lift;
+import frc.robot.subsystems.Lift.LiftPosition;
 
-public class IntakeSequence extends CommandBase {
+public class IntakeSequence extends SequentialCommandGroup {
   private Intake intake;
   private Lift lift;
 
-  public IntakeSequence(Intake intake, Lift lift) {
-    this.intake = intake;
-    this.lift = lift;
-  }
+  // deploy intake for specified amount of time
+  private ParallelRaceGroup deployIntake =
+      new RunCommand(intake::deploy, intake).withTimeout(Cal.Intake.AUTO_CLAMP_WAIT_TIME_SECONDS);
 
-  @Override
-  public void initialize() {
-    intake.deploy();
-    lift.setDesiredPosition(Lift.LiftPosition.GRAB_FROM_INTAKE);
-  }
+  // run intake, move lift to intake position, and open the grabber, until the robot sees an object
+  // and the lift is in position
+  private RunCommand runIntake = new RunCommand(intake::intakeGamePiece, intake);
+  private InstantCommand liftToIntakePos =
+      new InstantCommand(() -> lift.setDesiredPosition(LiftPosition.GRAB_FROM_INTAKE), lift);
+  private InstantCommand openGrabber = new InstantCommand(lift::openGrabber, lift);
+  private ParallelRaceGroup intakePrep =
+      new ParallelCommandGroup(runIntake, liftToIntakePos, openGrabber).until(lift::readyToIntake);
```

Comment:
I do not think the grabber can be opened at the same time that the lift is going to the intake position, but I could be wrong here.

Reply:
So I actually came up with a good solution for this, which you can see [here](https://github.com/cachemoney8096/2023-charged-up/commit/64f8c2). The `openGrabber()` and `closeGrabber()` functions don't actually open and close the grabber; they simply set a variable saying if the grabber is going to be open or closed. Then in `lift.periodic()` it checks that variable, and automatically opens it if the variable is true AND it's safe to do so. So what will happen here is that the grabber will open, close, and open again... which isn't that optimal and might get changed anyway, but not because of grabber safety.

Reply:
Ohhhhh wow I see. That is actually a really good way of doing it. When I saw this (without knowing what the function actually does, but just inferring from context and the comment) I thought that the functions `openGrabber` and `closeGrabber` actually set the grabber directly to open or closed instead of setting a variable. That is really cool!